educe/stac/edit/cmd/delete_anno.py

```diff
@@ -0,0 +1,112 @@
+# Author: Eric Kow
+# License: CeCILL-B (French BSD3-like)
+
+"""Delete an annotation"""
+
+from __future__ import print_function
+import sys
+
+from educe.stac.util.args import\
+    add_usual_input_args, add_usual_output_args,\
+    read_corpus, get_output_dir, announce_output_dir,\
+    anno_id
+from educe.stac.util.glozz import anno_id_from_tuple, anno_id_to_tuple
+from educe.stac.util.output import save_document
+
+
+
+def _is_match(wanted):
+    """
+    Given an annotation id, return a predicate that checks if
+    an annotation id matches
+    """
+    def pred(anno):
+        "curried second arg"
+        return anno_id_to_tuple(anno.local_id()) == wanted
+    return pred
+
+
+def _delete_in_doc(del_id, doc):
+    """Delete the annotations with the given id in the given document
+
+    NB: modifies doc
+    """
+    pretty_id = anno_id_from_tuple(del_id)
+    is_ok = lambda x: not _is_match(del_id)(x)
+    matches = [x for x in doc.annotations() if not is_ok(x)]
```

Comment:
I have tried to unroll the last 2 lines, plus the _is_match function, and arrived at the following formulation which, in my opinion, is really much much easier to understand and maintain:
```python
matches = [x for x in doc.annotations()
                  if anno_id_to_tuple(x.local_id()) == del_id]
```
Explanations could just happen in a comment line.

I have to say I don't feel really comfortable to try and favor certain coding styles, so won't feel really offended if you ignored this comment.